src/settings/lib/RuleEditor/rules-show-hint.js

```diff
@@ -782,11 +851,33 @@ class MultipleSelectionHintSection extends HintSection {
   }
 
   setList(list, selectedHintIndex = -1) {
+    this.updateSectionWidth();
+
     super.setList(list, selectedHintIndex);
 
+    this.filterItems(this.filterInput.value);
     this.setCompletionButtonVisibility(list.length > 1);
     this.setCompletionButtonEnabling(this.hasSelectedItems());
   }
+
+  updateSectionWidth() {
+    const headerStyle = getComputedStyle(this.header);
+
+    this.listContainer.style.width = headerStyle.width;
+    this.listContainer.style.margin = headerStyle.margin;
+  }
+
+  getListNodesArray() {
```

Comment:
Since here no function params are not needed I suggest making it as [getter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get).
I also suggest not mentioning the structure name as `Array` in this case. We want to be abstract on such things.
```suggestion
  get listNodes() {
```
so in code later it can be used without braces (e.g. `this.listNodes.length`).